examples/ota-requestor-app/linux/PersistentStorage.cpp

```diff
@@ -0,0 +1,251 @@
+/*
+ *   Copyright (c) 2021 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+#include "PersistentStorage.h"
+
+#include <lib/core/CHIPEncoding.h>
+#include <lib/support/Base64.h>
+#include <protocols/secure_channel/PASESession.h>
+
+#include <fstream>
+#include <memory>
+
+using String   = std::basic_string<char>;
+using Section  = std::map<String, String>;
+using Sections = std::map<String, Section>;
+
+using namespace ::chip;
+using namespace ::chip::Controller;
+using namespace ::chip::Logging;
+
+constexpr const char kFilename[]           = "/tmp/chip_tool_config.ini";
+constexpr const char kDefaultSectionName[] = "Default";
+constexpr const char kPortKey[]            = "ListenPort";
+constexpr const char kLoggingKey[]         = "LoggingLevel";
+constexpr const char kLocalNodeIdKey[]     = "LocalNodeId";
+constexpr const char kRemoteNodeIdKey[]    = "RemoteNodeId";
+constexpr LogCategory kDefaultLoggingLevel = kLogCategory_Detail;
+
+namespace {
+
+std::string StringToBase64(const std::string & value)
+{
+    std::unique_ptr<char[]> buffer(new char[BASE64_ENCODED_LEN(value.length())]);
+
+    uint32_t len =
+        chip::Base64Encode32(reinterpret_cast<const uint8_t *>(value.data()), static_cast<uint32_t>(value.length()), buffer.get());
+    if (len == UINT32_MAX)
+    {
+        return "";
+    }
+
+    return std::string(buffer.get(), len);
+}
+
+std::string Base64ToString(const std::string & b64Value)
+{
+    std::unique_ptr<uint8_t[]> buffer(new uint8_t[BASE64_MAX_DECODED_LEN(b64Value.length())]);
+
+    uint32_t len = chip::Base64Decode32(b64Value.data(), static_cast<uint32_t>(b64Value.length()), buffer.get());
+    if (len == UINT32_MAX)
+    {
+        return "";
+    }
+
+    return std::string(reinterpret_cast<const char *>(buffer.get()), len);
+}
+
+} // namespace
+
+CHIP_ERROR PersistentStorage::Init()
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    std::ifstream ifs;
+    ifs.open(kFilename, std::ifstream::in);
+    if (!ifs.good())
+    {
+        CommitConfig();
+        ifs.open(kFilename, std::ifstream::in);
+    }
+    VerifyOrExit(ifs.is_open(), err = CHIP_ERROR_OPEN_FAILED);
```

Comment:
usage of  return  macors would be prefferable: this does not  seem to have any exit cleanup code.

Reply:
It can be removed completely once we are able to not use `Controller` code here